### PR TITLE
Implement ID for operations that participate in money reservation

### PR DESCRIPTION
**Pull request by Artem-Semenov-dev**
This PR implements `OperationId`, which replaces `WithdrawalId`. These changes are provoked by the next PR that will implement shares purchase flow where not only `WalletWithdrawal` operation but another operation should be able to reserve money in the wallet.

### CODE REVIEW

File: model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java [hunk 1]
```diff
@@ -24,23 +24,28 @@
...
  */
 @Immutable
 @GeneratedMixin
-public interface WithdrawalEvent extends EventMessage {
+public interface MoneyReservationSignal extends EventMessage {
 
     /**
-     * Every event that participates in the withdrawal operation
-     * should have {@code WithdrawalId}.
+     * Every signal that participates in the money reservation operation
```
Comment (armiol) [inline model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java:43]: "Returns the ID of operation, in scope of which this signal is emitted."

File: model/src/main/proto/spine_examples/shareaware/identifiers.proto [hunk 2]
```diff
@@ -71,3 +71,16 @@ message WithdrawalId {
...
 }
+
+// Identifies a shares purchase process.
+message PurchaseId {
+    string uuid = 1 [(required) = true];
+}
+
+// Identifies the operation that interacts with the wallet by reserving money in it.
+message OperationId {
+    oneof id {
```
Comment (armiol) [inline model/src/main/proto/spine_examples/shareaware/identifiers.proto:82]: Let's name this field "type".

File: server/src/main/java/io/spine/examples/shareaware/server/wallet/WalletAggregate.java [hunk 3]
```diff
@@ -160,7 +160,9 @@ private void event(MoneyReservationCanceled e) {
                 .removeReservedMoney(withdrawalId);
     }
 
-    private static String extractWithdrawalIdValue(WithdrawalEvent e) {
-        return e.getWithdrawalProcess().getUuid();
+    private static String extractWithdrawalIdValue(MoneyReservationSignal e) {
```
Comment (armiol) [inline server/src/main/java/io/spine/examples/shareaware/server/wallet/WalletAggregate.java:163]: Please move this method to the signal interface, and shorten its name.

**Review by Vladyslav-Kuksiuk — COMMENTED** [review]
@Artem-Semenov-dev Generally LGTM, please see my comment.

File: model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java [hunk 4]
```diff
@@ -24,23 +24,45 @@
...
+     */
+    default WithdrawalId getWithdrawalProcess() {
+        return getOperation().getWithdrawal();
+    }
+
+    /**
+     * Retrieves the value of {@code WithdrawalId}
+     * without checking for {@code WithdrawalId} existence in {@code OperationId}.
+     *
+     * <p>In case when {@code WithdrawalId} is not existed in {@code OperationId} it will return an empty string.
```
Comment (armiol) [inline model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java:63]: "...is not set in the corresponding {@code OperationId}, returns an empty string."

File: model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java [hunk 5]
```diff
@@ -24,23 +24,45 @@
...
+     */
+    default String getWithdrawalIdValue() {
+        return getOperation().getWithdrawal().getUuid();
+    }
```
Comment (armiol) [inline model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java:67]: Let's also introduce a method telling about withdrawal:
```java
/* <documentation here> */
default boolean isPartOfWithdrawal() {
    // return getOperation().hasWithdrawal(); — something like this.
}
```

File: model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java [hunk 6]
```diff
@@ -24,23 +24,51 @@
...
+        return getOperation().getWithdrawal();
+    }
+
+    /**
+     * Retrieves the value of {@code WithdrawalId}
+     * without checking for {@code WithdrawalId} existence in {@code OperationId}.
+     *
+     * <p>In case when {@code WithdrawalId} is not set in the corresponding {@code OperationId}, returns an empty string.
+     */
+    default String getWithdrawalIdValue() {
```
Comment (armiol) [inline model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java:64]: Please omit "get" in those methods which we declare by hand. It's an old and obsolete practice.

**Review by armiol — APPROVED** [review]
LGTM with the respect of my comment about naming getters without "get".

**Pull request merged**